src/java/Files.java

```diff
@@ -443,30 +443,34 @@ public static int ReadStr(Files_FileDesc file, char[] str) {
     return i;
   }
 
-  public static int Seek(Files_FileDesc file, int pos) {
-    int r;
-    r = OK;
+  public static int Seek(Files_FileDesc file, long pos) {
+    file.err = OK;
     try {
       file.f.seek(pos);
-      if(file.err == EOF) {
-        file.err = OK;
-      }
     } catch(IOException e) {
-      r = IOERROR;
+      file.err = IOERROR;
     }
-    return r;
+    return file.err;
+  }
+
+  public static long Pos(Files_FileDesc file) {
```

Comment:
`oberonc` support only integers not longs, so we cannot call this from Oberon code.